src/ff/boolean_array.rs

```diff
@@ -138,13 +135,14 @@ macro_rules! boolean_array_impl {
                     (*self).into()
                 }
 
-                /// uses hashing in order to be compatible with larger array sizes
                 fn truncate_from<T: Into<u128>>(v: T) -> Self {
-                    let hk = Hkdf::<Sha256>::new(None, &v.into().to_le_bytes());
-                    let mut okm = [0u8; $bytes];
-                    //error invalid length from expand only happens when okm is very large
-                    hk.expand(&[], &mut okm).unwrap();
-                    <$name>::deserialize(&okm.into())
+                    let v = v.into();
+                    let mut val = Self::ZERO;
+                    for i in 0..$bits {
+                        val.set(i, Boolean::from((v >> i & 1) == 1));
```

Comment:
Much better :)

Reply:
But this is definitely needed somewhere else. It shouldn't be here for sure though. I'm going to assume @danielmasny is working on moving this into a different location.

Reply:
Is this the truncate function? It works this way however it does not lead to a secure PRSS/Multiply for arrays of more than 128 bits. We need to fix this as indicated in https://github.com/private-attribution/ipa/issues/812.